packages/destination-actions/src/destinations/adobe-target/adobeTarget_operations.ts

```diff
@@ -0,0 +1,60 @@
+import { RequestClient, IntegrationError } from '@segment/actions-core'
+
+function getNestedObjects(obj: { [x: string]: any }, objectPath = '', attributes: { [x: string]: string } = {}) {
+  Object.keys(obj).forEach((key) => {
+    const currObjectPath = objectPath ? `${objectPath}.${key}` : key
+    if (typeof obj[key] !== 'object' && obj[key]) {
+      attributes[currObjectPath] = obj[key].toString()
+    } else {
+      getNestedObjects(obj[key], currObjectPath, attributes)
+    }
+  })
+  return attributes
+}
+const objectToQueryString = (object: { [x: string]: { toString: () => string } }) =>
+  Object.keys(object)
+    .map((key) => `profile.${key}=${object[key].toString()}`)
+    .join('&')
+
+export default class AdobeTarget {
+  userId: string
+  clientCode: string
+  traits: object
+  request: RequestClient
+
+  constructor(userId: string, clientCode: string, traits: object, request: RequestClient) {
+    this.userId = userId
+    this.clientCode = clientCode
+    this.traits = traits
+    this.request = request
+  }
+
+  updateProfile = async () => {
+    const err = await this.lookupProfile(this.userId, this.clientCode)
+    if (err) {
+      throw err
+    } else {
+      const traits = getNestedObjects(this.traits)
+      const requestUrl = `https://${this.clientCode}.tt.omtrdc.net/m2/${
+        this.clientCode
+      }/profile/update?mbox3rdPartyId=${this.userId}&${objectToQueryString(traits)}`
+
+      return this.request(requestUrl, {
+        method: 'POST'
+      })
+    }
+  }
+
+  private lookupProfile = async (userId: string, clientCode: string): Promise<IntegrationError | undefined> => {
+    try {
+      await this.request(
+        `https://${clientCode}.tt.omtrdc.net/rest/v1/profiles/thirdPartyId/${userId}?client=${clientCode}`,
+        { method: 'get' }
+      )
+    } catch (error) {
+      return new IntegrationError('There is no profile found with this id', 'Profile not found', 404)
```

Comment:
@kdaswani Can you check if the Integration Error wording is okay. This won't show up in event tester, but will show up when adding a subscription and sending a test event. 
![image](https://user-images.githubusercontent.com/89420099/152543423-9eb2a6cd-6984-43dd-a468-043e6c22dd59.png)

Reply:
```suggestion
      return new IntegrationError('No profile found in Adobe Target with this mbox3rdPartyId', 'Profile not found', 404)
```
Made a slight change @rvadera12! Thanks for tagging me. What will show up in event tester and event delivery?

Reply:
Event tester responds with: 
![image](https://user-images.githubusercontent.com/89420099/152597915-af4b8bc5-e1f2-42cf-8b4d-f99dde70a36e.png)
This response comes from the app repo.